src-java/me/tonsky/persistent_sorted_set/Seq.java

```diff
@@ -156,4 +156,74 @@ public Seq rseq() {
     else
       return _set.slice(_keyTo, atEnd() ? null : first(), _cmp);
   }
-}
\ No newline at end of file
+
+  private static Object minKey (Leaf node) {
+    Leaf n = node;
+    while (n instanceof Node) n = ((Node) n)._children[0];
+    return n.minKey();
+  }
+
+  public Seq seek(Object to) { return seek(to, _cmp); }
+  public Seq seek(Object to, Comparator cmp) {
+    if (to == null) throw new RuntimeException("seek can't be called with a nil key!");
+
+    Seq seq = this._parent;
+    Leaf node = this._node;
+
+    if (_asc) {
+
+      while (node != null && cmp.compare(node.maxKey(), to) < 0){
+        if (seq == null) {
+          return null;
+        } else {
+          node = seq._node;
+          seq = seq._parent;
+        }
+      }
+
+      while (true) {
+        int idx = node.searchFirst(to, cmp);
+        if (idx < 0)
+          idx = -idx - 1;
+        if (idx == node._len)
+          return null;
+        if (node instanceof Node) {
+          seq = new Seq(null, this._set, seq, node, idx, null, null, true, _version);
+          node = seq.child();
+        } else { // Leaf
+          seq = new Seq(null, this._set, seq, node, idx, this._keyTo, cmp, true, _version);
+          return seq.over() ? null : seq;
+        }
+      }
+
+    } else {
+
+      while (cmp.compare(to, minKey(node)) < 0){
```

Comment:
This is a bit `awkward` here. There is currently no way to get the minimum key of any of it's descendants of a `Node` in constant time. This should probably be somehow made possible o/w `seek` for rseq will have a lot of unnecessary overhead.